src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -131,16 +155,16 @@ public function transform($entityOrCollection)
                 );
             }
         } else {
-            if (substr(\get_class($entityOrCollection), -1 * \strlen($this->className)) === $this->className) {
-                $collection = [$entityOrCollection];
-            } elseif ($entityOrCollection instanceof \ArrayAccess) {
+            if (substr(\get_class($value), -1 * \strlen($this->className)) === $this->className) {
```

Comment:
It should be `!$isArray && substr(\get_class($value), -1 * \strlen($this->className)) === $this->className`